src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/custom.py

```diff
@@ -0,0 +1,122 @@
+#---------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+#---------------------------------------------------------------------------------------------
+
+from azure.cli.core.commands import cli_command
+from azure.cli.core._util import CLIError
+
+from azure.cli.command_modules.acr.containerregistry.models import (
+    RegistryParameters,
+    RegistryProperties,
+    StorageAccountProperties
+)
+
+from ._factory import (
+    get_registry_service_client,
+    get_storage_end_point_suffix
+)
+
+from ._arm_utils import (
+    arm_get_registries_in_subscription,
+    arm_get_registries_in_resource_group,
+    arm_get_registry_by_name,
+    arm_deploy_template
+)
+
+from ._utils import (
+    get_registry_by_name,
+    get_resource_group_by_registry
+)
+
+from ._format import output_format
+
+def acr_list(resource_group=None):
+    '''Returns the list of container registries.
+    :param str resource_group: The name of resource group
+    '''
+    if resource_group:
+        return arm_get_registries_in_resource_group(resource_group)
+    else:
+        return arm_get_registries_in_subscription()
+
+def acr_create(resource_group, registry_name, location, #pylint: disable=too-many-arguments
+               storage_account_name=None, storage_account_key=None):
+    '''Returns the created container registry.
+    :param str resource_group: The name of resource group
+    :param str registry_name: The name of container registry
+    :param str location: The name of location
+    :param str storage_account_name: The name of storage account
+    :param str storage_account_key: The key of storage account
+    '''
+    if storage_account_name:
+        if storage_account_key:
+            storage_account_properties = \
+            StorageAccountProperties(name=storage_account_name,
```

Comment:
>storage_account_name [](start = 42, length = 20)

Storage account name, key, and connection string (which parses into name and key) can all be backed by environment variables and/or the config file. Since we support these within the storage module, customers would likely expect them to apply here. This code snippet is modified from the `validate_client_parameters` method in `storage._validators.py`

```
    from azure.cli.core._config import az_config
    from azure.cli.core.commands.validators import validate_key_value_pairs

    if not connection_string:
        connection_string = az_config.get('storage', 'connection_string', None)

    # if connection string supplied or in environment variables, extract account key and name
    if connection_string:
        conn_dict = validate_key_value_pairs(connection_string)
        account_name = conn_dict['AccountName']
        account_key = conn_dict['AccountKey']

    # otherwise, simply try to retrieve the remaining variables from environment variables
    if not storage_account_name:
        storage_account_name = az_config.get('storage', 'account', None)
    if not storage_account_key:
        storage_account_key = az_config.get('storage', 'key', None)
```